pxr/usdImaging/usdImaging/dataSourceCamera.cpp

```diff
@@ -109,6 +109,78 @@ class _Vec4fArrayToVec4dArrayDataSource
     typename HdTypedSampledDataSource<VtArray<GfVec4f>>::Handle _dataSource;
 };
 
+class _CameraExposureScaleDataSource
+    : public HdTypedSampledDataSource<float>
+{
+public:
+    HD_DECLARE_DATASOURCE(_CameraExposureScaleDataSource);
+
+    _CameraExposureScaleDataSource(
+        const SdfPath &sceneIndexPath,
+        UsdGeomCamera usdCamera,
+        const UsdImagingDataSourceStageGlobals &stageGlobals)
+    : _sceneIndexPath(sceneIndexPath)
+    , _usdCamera(usdCamera)
+    , _stageGlobals(stageGlobals)
+    {
+        static const HdDataSourceLocator exposureScaleLocator = 
+            HdCameraSchema::GetExposureScaleLocator();
+
+        static const std::vector<TfToken> inputNames = {
+            UsdGeomTokens->exposure,
+            UsdGeomTokens->exposureTime,
+            UsdGeomTokens->exposureIso,
+            UsdGeomTokens->exposureFStop,
+            UsdGeomTokens->exposureResponsivity
+        };
+
+        _inputs.resize(inputNames.size());
+        UsdPrim prim = _usdCamera.GetPrim();
+        for (TfToken inputName : inputNames) {
```

Comment:
@pmolodo If possible, avoid a copy of `TfToken` as it increases the ref count--  It should be safe to use `const TfToken&` or `const auto&` here.

Reply:
[avoid copying TfToken in loop](https://github.com/PixarAnimationStudios/OpenUSD/pull/3085/commits/7e94c68d78389b25c6fdab2fc1a56fc2ff657553)